stability: more information when timed mutex fires

tbg opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9769)

<!-- Reviewable:end -->

tamird commented:
Doesn't the second commit obviate the need for "name" from the first?

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 7 of 7 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[util/syncutil/timedmutex_test.go, line 33 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTNGs1VTMCi2h7zKBtQ:-KTNGs1WFkRYfedfifix:b3kre) ([raw file](https://github.com/cockroachdb/cockroach/blob/26bebb5c074a78c7bd311904884c21add246a9b7/util/syncutil/timedmutex_test.go#L33)):*
> ```Go
> 	var msgs []string
> 	syncutil.SetLogger(func(ctx context.Context, innerMsg string, args ...interface{}) {
> 		fmt.Fprintf(os.Stderr, innerMsg, args...)
> ```

?

---

*[util/syncutil/timedmutex_test.go, line 44 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTNGveO8oimhmkxo6B8:-KTNGveO8oimhmkxo6B9:b-c753bs) ([raw file](https://github.com/cockroachdb/cockroach/blob/26bebb5c074a78c7bd311904884c21add246a9b7/util/syncutil/timedmutex_test.go#L44)):*
> ```Go
> 		tm.Unlock()
> 
> 		if len(msgs) != 1 {
> ```

can you explicitly test for the content? i'd like to see it here for documentation's sake.

---

*[util/syncutil/timedmutex_test.go, line 56 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTNH-nd1Yi0rtCVnfYL:-KTNH-nd1Yi0rtCVnfYM:b-c753bs) ([raw file](https://github.com/cockroachdb/cockroach/blob/26bebb5c074a78c7bd311904884c21add246a9b7/util/syncutil/timedmutex_test.go#L56)):*
> ```Go
> 		tm.Unlock()
> 
> 		if len(msgs) != 1 {
> ```

can you explicitly test for the content? i'd like to see it here for documentation's sake.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-:-KTNHNx3sJrZzhB3p2VF:b4ngd50)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Sort of, I've made it clearer now by making the logger a callback and introducing metrics. The name is now gone. PTAL.

---

Review status: 0 of 11 files reviewed at latest revision, 3 unresolved discussions.

---

*[util/syncutil/timedmutex_test.go, line 33 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTNGs1VTMCi2h7zKBtQ:-KTNRLhL82cMOKLdnJmh:b-87j1qu) ([raw file](https://github.com/cockroachdb/cockroach/blob/26bebb5c074a78c7bd311904884c21add246a9b7/util/syncutil/timedmutex_test.go#L33)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ?
</details>
Gone.

---

*[util/syncutil/timedmutex_test.go, line 44 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTNGveO8oimhmkxo6B8:-KTNRSIqSAR01_l0ABDm:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/26bebb5c074a78c7bd311904884c21add246a9b7/util/syncutil/timedmutex_test.go#L44)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> can you explicitly test for the content? i'd like to see it here for documentation's sake.
</details>
Done.

---

*[util/syncutil/timedmutex_test.go, line 56 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTNH-nd1Yi0rtCVnfYL:-KTNRUY_vHKKcb8Wzkad:byo71nq) ([raw file](https://github.com/cockroachdb/cockroach/blob/26bebb5c074a78c7bd311904884c21add246a9b7/util/syncutil/timedmutex_test.go#L56)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> can you explicitly test for the content? i'd like to see it here for documentation's sake.
</details>
This is testing that no content has been added.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-:-KTNRDEY3WTYhnaXzrzb:bu0zlk)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 11 of 11 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[storage/metrics.go, line 238 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTOIfJwZllMat5iEyju:-KTOIfJwZllMat5iEyjv:b7r5ogc) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b15bbb79b7679cd5451d00c3a11444cc6e114a4/storage/metrics.go#L238)):*
> ```Go
> 		Help: "Number of successful intent resolutions"}
> 
> 	muReplicaNumLong = metric.Metadata{Name: "mutex.replica.long",
> ```

these names should all start with `meta`, I think.

---

*[storage/metrics.go, line 387 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTOJ7Y9KrgWMO9bbg7d:-KTOJ7Y9KrgWMO9bbg7e:b-gag6yf) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b15bbb79b7679cd5451d00c3a11444cc6e114a4/storage/metrics.go#L387)):*
> ```Go
> 
> 	// Mutex timing information.
> 	MuStoreNumLong   *metric.Counter
> ```

we don't have a "histogram" metric facility, do we?

---

*[storage/replica.go, line 511 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTOIRG_LfVCid4sKPO0:-KTOIRG_LfVCid4sKPO1:bpjjrjh) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b15bbb79b7679cd5451d00c3a11444cc6e114a4/storage/replica.go#L511)):*
> ```Go
> 
> 	raftMuLogger := func(ctx context.Context, msg string, args ...interface{}) {
> 		log.Warningf(ctx, "raftMu:"+msg, args...)
> ```

can you help me understand why this needs both the type aliases and these manual strings?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Gave this a Histogram makeover, PTAL at last commit (will squash and document properly if it looks good)

![image](https://cloud.githubusercontent.com/assets/5076964/19160410/c906546e-8bbd-11e6-81cd-64fee7c623af.png)

tamird commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Reviewed 10 of 10 files at r4.
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[storage/metrics.go, line 242 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPngqqN01Duq1Xuc3W:-KTPngqqN01Duq1Xuc3X:b-ev7ou0) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/metrics.go#L242)):*
> ```Go
> 
> 	metaMuReplicaTiming = metric.Metadata{Name: "mutex.replica",
> 		Help: "Duration of ReplicaMu critical sections"}
> ```

s/ReplicaMu/Replica mutex/

---

*[storage/metrics.go, line 244 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPnjpiD8O_-tKKaQXW:-KTPnjpiD8O_-tKKaQXX:b-l9lise) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/metrics.go#L244)):*
> ```Go
> 		Help: "Duration of ReplicaMu critical sections"}
> 	metaMuRaftTiming = metric.Metadata{Name: "mutex.raft",
> 		Help: "Duration of RaftMu critical sections"}
> ```

s/RaftMu/Replica Raft mutex/

---

*[storage/metrics.go, line 246 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPnlbdH0RlhCsT0WTr:-KTPnlbdH0RlhCsT0WTs:bofmdyz) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/metrics.go#L246)):*
> ```Go
> 		Help: "Duration of RaftMu critical sections"}
> 	metaMuStoreTiming = metric.Metadata{Name: "mutex.store",
> 		Help: "Duration of StoreMu critical sections"}
> ```

Store mutex

---

*[storage/metrics.go, line 533 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPouSXIJOb7rmoWsMr:-KTPouSXIJOb7rmoWsMs:b82d14n) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/metrics.go#L533)):*
> ```Go
> 
> 		// Mutex timing.
> 		MuReplicaTiming: metric.NewHistogram(
> ```

throughout: these names should have a unit: `MuStoreNanos`

---

*[storage/replica.go, line 515 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPoqoY1agPHcKB4mWL:-KTPoqoY1agPHcKB4mWM:bubcjdq) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/replica.go#L515)):*
> ```Go
> 		log.Warningf,
> 		func(t time.Duration) {
> 			r.store.metrics.MuRaftTiming.RecordValue(int64(t))
> ```

throughout: t.Nanoseconds()

---

*[util/syncutil/timedmutex.go, line 44 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPo0ca1jddv0r8Jv-l:-KTPo0ca1jddv0r8Jv-m:bky0fw6) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex.go#L44)):*
> ```Go
> 
> // ThresholdLogger returns a timing function which calls measure for each
> // duration and invokes the supplied print function (with the passed context)
> ```

s/duration/Lock-Unlock interval/

---

*[util/syncutil/timedmutex.go, line 45 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPo6V0OXpHQwYgnrkz:-KTPo6V0OXpHQwYgnrl-:bcomcg) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex.go#L45)):*
> ```Go
> // ThresholdLogger returns a timing function which calls measure for each
> // duration and invokes the supplied print function (with the passed context)
> // to dump an informative stack trace whenever the mutex is released after
> ```

and calls printf with ctx and a warning message containing a stack trace whenever ...

---

*[util/syncutil/timedmutex.go, line 50 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPoJgJVNtgbcj6tVIz:-KTPoJgJVNtgbcj6tVJ-:bfe6ood) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex.go#L50)):*
> ```Go
> 	ctx context.Context,
> 	warnDuration time.Duration,
> 	print func(context.Context, string, ...interface{}),
> ```

call this printf or even logf

---

*[util/syncutil/timedmutex_test.go, line 33 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPogM_PWhbkDV4aLLE:-KTPogM_PWhbkDV4aLLF:b-1szd9) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex_test.go#L33)):*
> ```Go
> 	var msgs []string
> 
> 	print := func(ctx context.Context, innerMsg string, args ...interface{}) {
> ```

nit: printf

---

*[util/syncutil/timedmutex_test.go, line 37 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPokQ8sUNMqbIS-Zmj:-KTPokQ8sUNMqbIS-Zmk:b-6q6m9n) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex_test.go#L37)):*
> ```Go
> 		msgs = append(msgs, formatted)
> 	}
> 	measure := func(time.Duration) {}
> ```

should actually test `measure` too

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-:-KTPpHeDDyLHEXHpwESz:b-uwkowd)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 4 of 11 files reviewed at latest revision, 13 unresolved discussions, some commit checks pending.

---

*[storage/metrics.go, line 238 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTOIfJwZllMat5iEyju:-KTQ4Jve39gylqRp7lcP:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9d6988678ba0d21c3d7b98b6f7beb60f63363df/storage/metrics.go#L238)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> these names should all start with `meta`, I think.
</details>
Done.

---

*[storage/metrics.go, line 387 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTOJ7Y9KrgWMO9bbg7d:-KTQ4ZO894ldaDI5OYvc:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9d6988678ba0d21c3d7b98b6f7beb60f63363df/storage/metrics.go#L387)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> we don't have a "histogram" metric facility, do we?
</details>
Done.

---

*[storage/metrics.go, line 242 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPngqqN01Duq1Xuc3W:-KTQ4WINM7PpkpQdschv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/metrics.go#L242)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/ReplicaMu/Replica mutex/
</details>
Done.

---

*[storage/metrics.go, line 244 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPnjpiD8O_-tKKaQXW:-KTQ4WjOUrhpEEziivv2:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/metrics.go#L244)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/RaftMu/Replica Raft mutex/
</details>
Done.

---

*[storage/metrics.go, line 246 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPnlbdH0RlhCsT0WTr:-KTQ4Wy5j8ApjzU6f2ae:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/metrics.go#L246)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Store mutex
</details>
Done.

---

*[storage/metrics.go, line 533 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPouSXIJOb7rmoWsMr:-KTQ5LqzqCzXSp0eGHig:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/metrics.go#L533)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> throughout: these names should have a unit: `MuStoreNanos`
</details>
Done.

---

*[storage/replica.go, line 511 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTOIRG_LfVCid4sKPO0:-KTP2DlKTetgIqisiFtP:b-t2sinw) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9d6988678ba0d21c3d7b98b6f7beb60f63363df/storage/replica.go#L511)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> can you help me understand why this needs both the type aliases and these manual strings?
</details>
You don't strictly require them, but I consider it an easy debugging improvement when the main log line includes the relevant information. For example, this easily allows me to grep for a part of this log line and I don't need to go through all of the stack traces to get an idea. However, at this point I removed them because we have the metrics.

---

*[storage/replica.go, line 515 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPoqoY1agPHcKB4mWL:-KTQC46GzOD4s7iByI1S:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/storage/replica.go#L515)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> throughout: t.Nanoseconds()
</details>
Done.

---

*[util/syncutil/timedmutex.go, line 44 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPo0ca1jddv0r8Jv-l:-KTQ6K9kugqGmRgveMFt:bfofge0) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex.go#L44)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/duration/Lock-Unlock interval/
</details>
Reworked a bit.

---

*[util/syncutil/timedmutex.go, line 45 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPo6V0OXpHQwYgnrkz:-KTQ7lTPPDieKSR3Dit5:b-qj4lsq) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex.go#L45)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> and calls printf with ctx and a warning message containing a stack trace whenever ...
</details>
Reworked.

---

*[util/syncutil/timedmutex.go, line 50 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPoJgJVNtgbcj6tVIz:-KTQ7t5JSyn4vV8rvG_6:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex.go#L50)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> call this printf or even logf
</details>
Done.

---

*[util/syncutil/timedmutex_test.go, line 33 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPogM_PWhbkDV4aLLE:-KTQ808hgJZ_lYqdGWXZ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex_test.go#L33)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> nit: printf
</details>
Done.

---

*[util/syncutil/timedmutex_test.go, line 37 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTPokQ8sUNMqbIS-Zmj:-KTQ80XS2e0zznvKyvM_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/82732afac08c9edb27a50255f8734cdea3155cdf/util/syncutil/timedmutex_test.go#L37)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> should actually test `measure` too
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769)*
<!-- Sent from Reviewable.io -->

tamird commented:
Awesome.

---

Reviewed 7 of 7 files at r5.
Review status: 10 of 11 files reviewed at latest revision, 2 unresolved discussions.

---

*[util/syncutil/timedmutex_test.go, line 61 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTQHmK9a9Q3VvLLgl4n:-KTQHmK9a9Q3VvLLgl4o:b3tjvga) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e658fd44ad981601b4974be9d36ef0bc074dc69/util/syncutil/timedmutex_test.go#L61)):*
> ```Go
> 
> 	numMeasurements = 0
> 	msgs = []string{}
> ```

nit: `msgs = nil`

---

*[util/syncutil/timedmutex_test.go, line 78 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTQHeLD4ll1Hoj4yYCg:-KTQHeLEYO0rk3IIkMtT:b219l7r) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e658fd44ad981601b4974be9d36ef0bc074dc69/util/syncutil/timedmutex_test.go#L78)):*
> ```Go
> 
> 		if len(msgs) != 0 {
> 			t.Fatalf("mutex warned erroneously: %+v", msgs[1:])
> ```

remove the slicing here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-:-KTQHsSmFc71qI_8-Kiz:bccpe8g)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 7 of 11 files reviewed at latest revision, 2 unresolved discussions.

---

*[util/syncutil/timedmutex_test.go, line 61 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTQHmK9a9Q3VvLLgl4n:-KTQINHw28zdFycW4g1q:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e658fd44ad981601b4974be9d36ef0bc074dc69/util/syncutil/timedmutex_test.go#L61)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> nit: `msgs = nil`
</details>
Done.

---

*[util/syncutil/timedmutex_test.go, line 78 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769#-KTQHeLD4ll1Hoj4yYCg:-KTQINYGVTsikMjgrBJg:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e658fd44ad981601b4974be9d36ef0bc074dc69/util/syncutil/timedmutex_test.go#L78)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> remove the slicing here.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769)*
<!-- Sent from Reviewable.io -->

tbg committed:
stability: metrics and better messages for TimedMutex

Disambiguate mutexes in stack traces (through a wrapper type).
Record mutex critical section durations to metrics, exposing the maximum over
the last minute to the ui.

tamird commented:
Reviewed 5 of 5 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9769)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Green at least, TFTR!

maxlang commented on ui/app/containers/nodeGraphs.tsx:
```diff
@@ -273,6 +273,24 @@ export default class extends React.Component<IInjectedProps, {}> {
+                <Metric name="cr.store.mutex.storenanos-max" title="StoreMu"
+                        aggregateMax downsampleMax />
+              </Axis>
+              <Axis format={ (n: number) => d3.format(".1f")(NanoToMilli(n)) } label="Milliseconds">
```
Missed this when it came through. Currently the graphs only supports 1 axis with multiple metrics, so the axes after the first are ignored.

tschottdorf replied:
Whoops, not sure what I did there. Certainly didn't want more than one axis. https://github.com/cockroachdb/cockroach/pull/9835
